api/envoy/config/route/v3/route_components.proto

```diff
@@ -911,6 +911,10 @@ message RouteAction {
     // in the
     // :ref:`envoy_v3_api_field_config.core.v3.TypedExtensionConfig.name` field.
     string cluster_specifier_plugin = 37;
+
+    // Custom cluster provider configuration to use to determine the cluster for requests on this route.
+    core.v3.TypedExtensionConfig cluster_provider = 39
```

Comment:
You are right. And I am waiting #20301 to be merged, then I will update this pr. Because maybe I need to move `Cluster SpecifierPlugin` to `route_component.proto` from `route.proto`.

Reply:
Yeah, I agree with using the original type for clarity in this case.